bounce/server/api/util.py

```diff
@@ -0,0 +1,141 @@
+"""Utility functions unsed by Bounce endpoints."""
+
+import hashlib
+import re
+from datetime import datetime, timedelta
+
+import bcrypt
+from jose import jwt
+
+# Regexes for validating passwords and usernames
+# pylint: disable=anomalous-backslash-in-string
+UPPERCASE = re.compile('[A-Z]+')
+LOWERCASE = re.compile('[a-z]+')
+NUMBERS = re.compile('[0-9]+')
+PASSWORD_SYMBOLS = re.compile('[\.\-!@#$%^&*?_+ ]+')
+USERNAME_SYMBOLS = re.compile('[\.\-_]+')
+
+# All the regexes that a valid password must match
+PASSWORD_REGEXES = [UPPERCASE, LOWERCASE, NUMBERS, PASSWORD_SYMBOLS]
+USERNAME_REGEXES = [UPPERCASE, LOWERCASE, NUMBERS, USERNAME_SYMBOLS]
+
+# Username and password length restrictions
+MIN_PASSWORD_LENGTH = 8
```

Comment:
should there be a max password too?

Reply:
I left off the max because I figured people should be able to make the passwords as long as they want.